seismicpro/gather/gather.py

```diff
@@ -1398,20 +1419,42 @@ def _plot_histogram(self, ax, title, x_ticker, y_ticker, x_tick_src="amplitude",
 
     # pylint: disable=too-many-arguments
     def _plot_seismogram(self, ax, title, x_ticker, y_ticker, x_tick_src=None, y_tick_src='time', colorbar=False,
-                         q_vmin=0.1, q_vmax=0.9, event_headers=None, top_header=None, **kwargs):
+                         q_vmin=0.1, q_vmax=0.9, event_headers=None, top_header=None, masks=None, **kwargs):
         """Plot the gather as a 2d grayscale image of seismic traces."""
         # Make the axis divisible to further plot colorbar and header subplot
         divider = make_axes_locatable(ax)
         vmin, vmax = self.get_quantile([q_vmin, q_vmax])
         kwargs = {"cmap": "gray", "aspect": "auto", "vmin": vmin, "vmax": vmax, **kwargs}
         img = ax.imshow(self.data.T, **kwargs)
+        if masks is not None:
+            default_mask_kwargs = {"aspect": "auto", "alpha": 0.5, "interpolation": "none", "color": "red"}
+            for mask_kwargs in self._process_masks(masks):
+                mask_kwargs = {**default_mask_kwargs, **mask_kwargs}
+                mask = mask_kwargs.pop("masks")
+                if np.nansum(mask) == 0:
+                    continue
```

Comment:
If this is empty mask check, consider
```suggestion
                if np.isnan(mask).all():
                    continue
```
Works properly in case `mask = [-1, np.nan, 1]`

Reply:
@anton-br If you deside fixing this consider also moving this if to `_process_masks`